### PR TITLE
Fix golint failures of e2e/framework/providers/gce/recreate_node.go

### PR DESCRIPTION
**Pull request by atoato88**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

This fixes golint failures of the following file:

test/e2e/framework/providers/gce/recreate_node.go

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

Ref: #68026

### CODE REVIEW

**Comment by oomichi** [comment]
@atoato88 Thanks for doing this. Could we remove the file from `hack/.golint_failures` ?

/ok-to-test

**Comment by atoato88** [comment]
@oomichi I've push a commit that removes `test/e2e/framework/providers/gce` from the `hack/.golint_failures`

File: test/e2e/framework/providers/gce/recreate_node.go [hunk 1]
```diff
@@ -39,19 +39,19 @@ func nodeNames(nodes []v1.Node) []string {
...
 	var originalPodNames []string
 	var ps *testutils.PodStore
 	systemNamespace := metav1.NamespaceSystem
-	BeforeEach(func() {
+	ginkgo.BeforeEach(func() {
 		framework.SkipUnlessProviderIs("gce", "gke")
 		var err error
 		numNodes, err := framework.NumberOfRegisteredNodes(f.ClientSet)
-		Expect(err).NotTo(HaveOccurred())
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
```
Comment (oomichi) [inline test/e2e/framework/providers/gce/recreate_node.go]: Oops, sorry for missing this in my previous comment.
Can we replace here with just `framework has ExpectNoError(err)` ?

File: test/e2e/framework/providers/gce/recreate_node.go [hunk 2]
```diff
@@ -39,19 +39,19 @@ func nodeNames(nodes []v1.Node) []string {
...
 		originalNodes, err = framework.CheckNodesReady(f.ClientSet, numNodes, framework.NodeReadyInitialTimeout)
-		Expect(err).NotTo(HaveOccurred())
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
```
Comment (oomichi) [inline test/e2e/framework/providers/gce/recreate_node.go]: ditto

File: test/e2e/framework/providers/gce/recreate_node.go [hunk 3]
```diff
@@ -69,13 +69,13 @@ var _ = Describe("Recreate [Feature:Recreate]", func() {
...
-		if CurrentGinkgoTestDescription().Failed {
+	ginkgo.AfterEach(func() {
+		if ginkgo.CurrentGinkgoTestDescription().Failed {
 			// Make sure that addon/system pods are running, so dump
 			// events for the kube-system namespace on failures
-			By(fmt.Sprintf("Collecting events from namespace %q.", systemNamespace))
+			ginkgo.By(fmt.Sprintf("Collecting events from namespace %q.", systemNamespace))
 			events, err := f.ClientSet.CoreV1().Events(systemNamespace).List(metav1.ListOptions{})
-			Expect(err).NotTo(HaveOccurred())
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
```
Comment (oomichi) [inline test/e2e/framework/providers/gce/recreate_node.go]: ditto

File: test/e2e/framework/providers/gce/recreate_node.go [hunk 4]
```diff
@@ -104,7 +104,7 @@ func testRecreate(c clientset.Interface, ps *testutils.PodStore, systemNamespace
 	}
 
 	nodesAfter, err := framework.CheckNodesReady(c, len(nodes), framework.RestartNodeReadyAgainTimeout)
-	Expect(err).NotTo(HaveOccurred())
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
```
Comment (oomichi) [inline test/e2e/framework/providers/gce/recreate_node.go]: ditto

File: test/e2e/framework/providers/gce/recreate_node.go [hunk 5]
```diff
@@ -115,7 +115,7 @@ func testRecreate(c clientset.Interface, ps *testutils.PodStore, systemNamespace
 	// Make sure the pods from before node recreation are running/completed
 	podCheckStart := time.Now()
 	podNamesAfter, err := framework.WaitForNRestartablePods(ps, len(podNames), framework.RestartPodReadyAgainTimeout)
-	Expect(err).NotTo(HaveOccurred())
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
```
Comment (oomichi) [inline test/e2e/framework/providers/gce/recreate_node.go]: ditto

File: test/e2e/framework/providers/gce/recreate_node.go (hunk 1 above)
Comment (atoato88) [reply test/e2e/framework/providers/gce/recreate_node.go]: done.

File: test/e2e/framework/providers/gce/recreate_node.go (hunk 2 above)
Comment (atoato88) [reply test/e2e/framework/providers/gce/recreate_node.go]: done.

File: test/e2e/framework/providers/gce/recreate_node.go (hunk 3 above)
Comment (atoato88) [reply test/e2e/framework/providers/gce/recreate_node.go]: done.

File: test/e2e/framework/providers/gce/recreate_node.go (hunk 4 above)
Comment (atoato88) [reply test/e2e/framework/providers/gce/recreate_node.go]: done.

File: test/e2e/framework/providers/gce/recreate_node.go (hunk 5 above)
Comment (atoato88) [reply test/e2e/framework/providers/gce/recreate_node.go]: done. 🐶

**Review by neolit123 — COMMENTED** [review]
thanks
/lgtm
/approve
/kind cleanup
/priority backlog

**Comment by neolit123** [comment]
@atoato88 
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/77034/pull-kubernetes-verify/1121632730170265607
please run `./hack/update-bazel.sh` and force push the change in the same commit.
thanks

**Review by neolit123 — COMMENTED** [review]
/approve

**Pull request merged**